### PR TITLE
Add API tests to cmake builds

### PR DESCRIPTION
**Pull request by khaledhosny**
CMake is not my thing, but this seems to work.

### CODE REVIEW

**Comment by khaledhosny** [comment]
This is mainly trying to address https://github.com/harfbuzz/harfbuzz/issues/588, but we still need to enable glib for the VS builds to get any tests and I have no idea how.

**Comment by ebraminio** [comment]
Great work Khaled, thanks. AFAIK just cross-compile configs on circleci don't have the chance to run binary on a simulated machine so running tests are not possible there.

> but we still need to enable glib for the VS builds to get any tests and I have no idea how.

You can install glib with vcpkg, https://github.com/behdad/glyphy/blob/master/appveyor.yml

**Comment by ebraminio** [comment]
I guess I turned on CircleCI on PRs, please send another commit to see.

**Comment by ebraminio** [comment]
`vcpkg` doesn't need to be installed as https://github.com/appveyor/ci/issues/1076 and https://www.appveyor.com/docs/lang/cpp/#vc-packaging-tool

File: .circleci/config.yml [hunk 1]
```diff
@@ -51,47 +54,59 @@ jobs:
       - checkout
       - run: apt update && apt install ragel
```
Comment (ebraminio) [inline .circleci/config.yml]: glib and rest of packages installed on travisci should be installed here for tests I guess.

**Comment by ebraminio** [comment]
Ok, #643 is able to enable glib and freetype on appveyor but it causes long delay on builds which parts of it will be cached. I guess however you should consider disabling tests on ARM.

File: appveyor.yml [hunk 2]
```diff
@@ -31,8 +31,9 @@ build_script:
   - 'if "%compiler%"=="cmake" md build'
   - 'if "%compiler%"=="cmake" cd build'
   - 'if "%compiler%"=="cmake" set PATH=%PATH%;C:\Program Files (x86)\MSBuild\14.0\Bin;c:\msys64\mingw64\bin' # msys2 is added just for having "ragel" on PATH
-  - 'if "%compiler%"=="cmake" cmake -DHB_HAVE_UNISCRIBE=ON -DHB_HAVE_DIRECTWRITE=ON -G "%generator%" ../'
+  - 'if "%compiler%"=="cmake" cmake -DHB_HAVE_UNISCRIBE=ON -DHB_HAVE_DIRECTWRITE=ON -DHB_HAVE_GLIB=ON -G "%generator%" ../'
```
[the remark this answers is not in the document]
Comment (ebraminio) [reply appveyor.yml]: Merged

**Comment by ebraminio** [comment]
glib is enabled on appveyor x86 and x64, you should rebase appveyor config

**Comment by ebraminio** [comment]
Oh, please don't give up on rest of not cross-compile circleci configs. "base" is a good candidate for making a full featured cmake build on Linux using Debian packages, "oracledeveloperstudio" also can be considered for that. Also you can use CircleCI's rerun job with SSH for testing the config before changing the PR.

**Comment by khaledhosny** [comment]
I just wanted to simplify this PR and do the others in a new one.

**Comment by ebraminio** [comment]
```diff
diff --git a/.circleci/config.yml b/.circleci/config.yml
index 370252a..682fb86 100644
--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -41,8 +41,9 @@ jobs:
       - image: dockcross/base
     steps:
       - checkout
-      - run: apt update && apt install ragel
-      - run: cmake -Bbuild -H. -GNinja && ninja -Cbuild
+      - run: apt update && apt install pkg-config ragel gtk-doc-tools libfreetype6-dev libglib2.0-dev libcairo2-dev libicu-dev libgraphite2-dev
+      - run: cmake -DHB_HAVE_FREETYPE=ON -DHB_HAVE_GRAPHITE2=ON -DHB_BUILTIN_UCDN=ON -DHB_HAVE_GLIB=ON -DHB_HAVE_ICU=ON -DHB_BUILD_UTILS=ON -Bbuild -H. -GNinja
+      - run: ninja -Cbuild && ninja install
 
   psvita:
     docker:
```

enables most of cmake on "base". I believe you can ship this along enabling ctest on it.

**Comment by ebraminio** [comment]
Ops, "      - run: ninja -Cbuild && ninja install" doesn't have ctest part and "-y" flag is missed on `apt install`, sorry for both

**Comment by ebraminio** [comment]
This looks great Khaled. I can add cmake install later. So please merge it whenever you see fit. Thanks

**Pull request merged**